File: frontend/src/services/resourceService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/resources';

export const getResources = async (filters = {}) => {
    const params = new URLSearchParams();
    if (filters.type) params.append('type', filters.type);
    if (filters.capacity) params.append('capacity', filters.capacity);
    if (filters.location) params.append('location', filters.location);
    
    const response = await axios.get(API_URL, { params });
    return response.data;
};

export const createResource = async (resource) => {
    const response = await axios.post(API_URL, resource);
    return response.data;
};

export const updateResource = async (id, resource) => {
    const response = await axios.put(`${API_URL}/${id}`, resource);
    return response.data;
};

export const deleteResource = async (id) => {
    const response = await axios.delete(`${API_URL}/${id}`);
    return response.data;
};

File: backend/src/main/java/SwiftFix/backend/service/ResourceService.java
package SwiftFix.backend.service;

import SwiftFix.backend.exception.ResourceNotFoundException;
import SwiftFix.backend.model.Resource;
import SwiftFix.backend.repository.ResourceRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class ResourceService {

    private final ResourceRepository resourceRepository;

    public Resource createResource(Resource resource) {
        if (resource.getStatus() == null) {
            resource.setStatus("ACTIVE");
        }
        return resourceRepository.save(resource);
    }

    public List<Resource> getResources(String type, Integer capacity, String location) {
        return resourceRepository.findWithFilters(type, capacity, location);
    }

    public Resource getResourceById(Long id) {
        return resourceRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Resource not found with id: " + id));
    }

    public Resource updateResource(Long id, Resource resourceDetails) {
        Resource resource = getResourceById(id);
        
        if (resourceDetails.getName() != null) resource.setName(resourceDetails.getName());
        if (resourceDetails.getType() != null) resource.setType(resourceDetails.getType());
        if (resourceDetails.getCapacity() != null) resource.setCapacity(resourceDetails.getCapacity());
        if (resourceDetails.getLocation() != null) resource.setLocation(resourceDetails.getLocation());
        if (resourceDetails.getAvailabilityWindows() != null) resource.setAvailabilityWindows(resourceDetails.getAvailabilityWindows());
        if (resourceDetails.getStatus() != null) resource.setStatus(resourceDetails.getStatus());
        
        return resourceRepository.save(resource);
    }

    public void deleteResource(Long id) {
        Resource resource = getResourceById(id);
        resourceRepository.delete(resource);
    }
}
